scipy/ndimage/filters.py

```diff
@@ -618,6 +618,8 @@ def _correlate_or_convolve(input, weights, output, mode, cval, origin,
     if not weights.flags.contiguous:
         weights = weights.copy()
     output = _ni_support._get_output(output, input)
+    if hasattr(mode, '__iter__') and not isinstance(mode, str):
```

Comment:
Rather than looking for double-underscore attributes it might be cleaner to use:
```
from collections.abc import Iterable
...

if not isinstance(mode, str) and isinstance(mode, Iterable):
    ...
```

Reply:
That sounds good to me.  I copied the style used here from elsewhere in ndimage, so we could make the same change there as well:
https://github.com/scipy/scipy/blob/7a73b1eb63b899b4e4772c4d3e348f4c2a6ddd57/scipy/ndimage/_ni_support.py#L59